pkg/operator/ceph/csi/controller.go

```diff
@@ -299,16 +310,44 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 		if err != nil {
 			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to update CSI driver options for cluster %q", cluster.Name)
 		}
-	}
 
-	err = r.validateAndConfigureDrivers(serverVersion, ownerInfo)
-	if err != nil {
-		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to configure ceph csi")
+		// Skip the new CSI-operator creation when holder pod is enabled until multus support is added in the CSI operator
+		if EnableCSIOperator() {
+			err := r.stopDrivers(serverVersion)
```

Comment:
I'll think about this how to separate between two and delete only the one created by rook

Reply:
hmm, currently I don't see any specific labels or annotations to differentiate between the two. @Madhu-1 @travisn any suggestions?

Reply:
What resources exactly need to be deleted when transitioning from the old CSI driver to the new? There are some resources that need to be deleted (e.g. provisioner and node plugins since they have different pod names than the new ones), but there are other resources that should not be deleted since they will have the same name. Can we just delete the resources that have different names than what is expected to be named by the CSI operator? Then we don't need to worry about checking owner references, etc.

Reply:
The deployment, daemonset, service need to be deleted and we also need to delete the kubernetes csidriver object as the new operator is not taking ownership of the existing object it might create a problem if we try to take ownership, as we might end up taking unexpected ownerships as well. @nb-ohad anything to add here?

Reply:
@Madhu-1 With the k8s csidriver resource, rook does not add an ownerref because it is a cluster-wide resource, so how can the ownership be examined in that case? It seems simple enough that Rook can just skip deleting it and CSI operator can start reconciling it. 

And which service are you referring to that needs deletion? `csi-rbdplugin-metrics` and `csi-cephfsplugin-metrics`? Will those have the same name when owned by the csi operator?

Reply:
@travisn The name of the k8s CSI driver is a known name. It is just a matter of creating a CSI OP Driver with the same name and attaching our ownership annotation to it (without it we will assume someone else created the k8s driver and we fail the reconcile). The ownership annotation is just a NamespacedName serialized to JSON.

Reply:
@travisn I am not sure what you are referring to. But nevertheless, I will try to answer:

One of the products of a `driver.csi.ceph.io` reconciliation is the creation of a matching k8s CSI driver CR (with the same name). But we need to consider the fact that the driver name we choose right already be claimed by another driver - either by a driver that was deployed using the operator in another namespace or by a manual deployment of a k8s CSI driver CR (e.g. using helm or some other means).

To counter that, we introduced a validation step at the start of our reconcile function that fails the reconciliation if a k8s CSI driver already exists on the cluster with the same name, and we can prove that this driver is owned by the `driver.csi.ceph.io` we are reconciling. 

The way we achieve that is by marking each k8s CSI driver we create with an "owner annotation" that contains the namespace name of the `driver.csi.ceph.io` that reconciled it. Then we will allow to continue the reconciliation in one of two cases: 
1. We cannot find a k8s CSI driver with the same name
2. We found a k8s CSI driver with the same name, but it has an annotation that shows that the current reconciled `driver.csi.ceph.io` is its "owner"

Reply:
My intent is really just to smooth the transition from rook ownership to the csi operator ownership of the driver.
What if Rook adds the csi operator annotation to the csidriver? Then the csi operator will immediately assume ownership. 

The alternative is for rook to delete its csidriver and then csi operator creates the new one that it owns. But that will have some short time period where the csi driver has disappeared and would prevent some operations during upgrade. 

On a related note, similarly, rook should potentially delay deleting the node and controller plugins until the new plugins start. There will certainly be some downtime during this transition where some csi operations would be failing or delayed, but we at least need to minimize that.

Reply:
I see,
My understanding is that deletion of the k8s CSI Driver might have some unwanted side effects. This is why we came up with a plan for the transition of ownership in the first place. If there are no side effects and it is safe to delete the k8s CSI driver then I agree: we should follow the simplest solution and delete it. 

Regarding the potential side effects, I will leave the explanation to @Madhu-1

Reply:
Others depend on this object (UI/KUBELET) etc, it's fair to delete the one operator is creating as it's one time not always.

Reply:
@Madhu-1 @nb-ohad Can you give more details about the plan for transition of ownership that minimizes the downtime? I'd like to understand this better, but as a separate discussion is fine so we don't block this PR.